### ISSUE TITLE
Expose node-specific config/options via RPC

### ISSUE THREAD

**Issue by jacogr** [paritytech/substrate#4965]
Some of the information that would make sense -

- the node type (validator, light client, etc.)
- the pruning mode (archive or number of blocks)
- execution mode (interpreted or compiled)
- ... there may be a flag (or two) from the cli that is useful and that I am missing

We may turning this into a hold-all and add elsewhere duplicated info such as system_name and system_version as well.

**Comment by tomaka** [comment]
> the node type (validator, light client, etc.)

This one already exists:
https://github.com/paritytech/substrate/blob/0227ff513a045305efee0d8c2cb6025bcefb69ff/client/rpc-api/src/system/mod.rs#L103-L105

**Comment by jacogr** [comment]
Agreed, that one does exist - the one above assumes we can combine all that info, so instead of making multiple calls, there is one single RPC that just gives you all at once. (Hence also mentioning name/version above, this is also elsewhere)
